qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +292,51 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler is not None:
+                qc = self.construct_circuit(amplification_problem, power, measurement=True)
+                job = self._sampler.run([qc])
+
+                try:
+                    results = job.result()
+                except Exception as exc:
+                    raise AlgorithmError("Sampler job failed.") from exc
+
                 num_bits = len(amplification_problem.objective_qubits)
+                circuit_results = {
+                    np.binary_repr(k, num_bits): v for k, v in results.quasi_dists[0].items()
+                }
+                top_measurement, max_probability = max(circuit_results.items(), key=lambda x: x[1])
+                shots = self._sampler.run_options.get("shots")
```

Comment:
This line getting shots - I don't think it used right - shots are in quantum_instance part but not here. Should it just be removed.

Reply:
Thanks. Right, I forgot to remove.